### PR TITLE
Edits to tutorial

### PR DESCRIPTION
**Pull request by stites**
I was just going through this and tried to do some clean up, I wound up changing a bit more than I thought, there's also a Moore-pun towards the end — feel free to tell me to drop any of it!

I've inlined the comments so that this PR is a little easier to go through.

### CODE REVIEW

File: Main.hs [hunk 1]
```diff
@@ -381,7 +388,10 @@ m1TransitionFmB _ = Moore M1A m1TransitionFmA
 m1 :: Moore Char M1State
 m1 = Moore M1A m1TransitionFmA
 
--- Turn the Moore state machine into a process
+-- Turn the Moore state machine into a process.
+--
+-- Recall the definition of a process:
+-- > type ProcessT m a b = MachineT m (Is a) b
```
Comment (stites) [inline Main.hs:394]: restate how `Process` is defined (it's referenced, but not used)

File: Main.hs [hunk 2]
```diff
@@ -412,9 +422,6 @@ state, and outputs the char used to transition.
 
 -}
 
-m2Mealy :: Char -> (M1State, Mealy Char M1State)
-m2Mealy = m2TransitionFmA
```
Comment (stites) [inline Main.hs:416]: remove unneeded m2Mealy

File: Main.hs [hunk 3]
```diff
@@ -352,20 +352,27 @@ Construct a Moore machine from a state valuation and transition function
...
+In the following examples, we use @auto@ which comes from the
+@'Data.Machine.Process.Automaton'@ typeclass:
 
+@
+  auto :: k a b -> Process a b
+@
+
+This means that we can take some routing function, @k a b@, and lift it
+into a process. You can think of this as a specific @'construct'@ for
+automatons.
```
Comment (stites) [inline Main.hs:364]: explain `auto` before starting the state machines section

File: Main.hs [hunk 4]
```diff
@@ -352,20 +352,27 @@ Construct a Moore machine from a state valuation and transition function
...
+-- For this next section, let's have two states and construct a simple state
+-- machine the state flips every time there is a new input. We'll use this
+-- state machine for the first examples of Moore and Mealy machines.
 
-Construct a simple Moore machine
---------------------------------
-
-This will have two states, and every time there is an input it changes
-state.
+data M1State = M1A | M1B deriving (Eq,Show,Ord)
```
Comment (stites) [inline Main.hs:371]: move states out of Moore Machine section since they are used later on in Mealy

File: Main.hs [hunk 5]
```diff
@@ -448,31 +455,25 @@ m2m = (source "abcde") ~> m2a
 -- When they run we only see the state as output.
 
 {-
-A Moore machine can be defined as a 6-tuple ( S, S0, Σ, Λ, T, G ) consisting of the following:
+Mealy and Moore machines can both be represented by a 6-tuple, ( S, S0, Σ, Λ, T, G ), consisting of the following:
```
Comment (stites) [inline Main.hs:458]: get rid of redundancies with the 6-tuple explanation.

File: Main.hs [hunk 6]
```diff
@@ -583,22 +584,34 @@ m4m = (source [I0,I0,I0,I1,I1,I1,I0,I0]) ~> m4a
...
--- | Construct a Moore machine from a state valuation and transition function
-unfoldMoore1 :: (s -> (b, a -> s)) -> s -> Moore a b
-unfoldMoore1 f = go where
-  go s = case f s of
-    (b, g) -> Moore b (go . g)
-{-# INLINE unfoldMoore1 #-}
-
 {-
+understanding unfoldMoore
+-------------------------
```
Comment (stites) [inline Main.hs:589]: breaking up the unfoldMoore signature in your explanation — also putting the definition in a comment

File: Main.hs [hunk 7]
```diff
@@ -614,9 +627,15 @@ fMoore :: M1State -> (M1State, XIn -> M1State)
 fMoore M1A = (M1A,m6Fm1A)
 fMoore M1B = (M1B,m6Fm1B)
 
+
+-- So here, we can see that we can lift a function with the
+-- shape of a Moore, into a Moore, if we have all the right
+-- pieces
```
Comment (stites) [inline Main.hs:633]: comment/reiteration on what unfolding a moore machine does.

**Comment by alanz** [comment]
Thanks

**Pull request merged**